subgraphs/exchange/exchange.graphql

```diff
@@ -93,6 +93,105 @@ type SubsidizedSwapped @entity {
   path: [Bytes!]
 }
 
+# For LimitOrder contract
+type LimitOrderFilledByProtocol @entity(immutable: true) {
+  id: ID!
+  orderHash: Bytes! # bytes32
+  maker: Bytes! # address
+  taker: Bytes! # address
+  allowFillHash: Bytes! # bytes32
+  relayer: Bytes! # address
+  profitRecipient: Bytes! # address
+  fillReceiptMakerToken: Bytes! # address
+  fillReceiptTakerToken: Bytes! # address
+  fillReceiptMakerTokenFilledAmount: BigInt! # uint256
+  fillReceiptTakerTokenFilledAmount: BigInt! # uint256
+  fillReceiptRemainingAmount: BigInt! # uint256
+  fillReceiptMakerTokenFee: BigInt! # uint256
+  fillReceiptTakerTokenFee: BigInt! # uint256
+  relayerTakerTokenProfit: BigInt! # uint256
+  relayerTakerTokenProfitFee: BigInt! # uint256
+  blockNumber: BigInt!
+  blockTimestamp: BigInt!
+  transactionHash: Bytes!
+}
+
+type LimitOrderFilledByTrader @entity(immutable: true) {
+  id: ID!
+  orderHash: Bytes! # bytes32
+  maker: Bytes! # address
+  taker: Bytes! # address
+  allowFillHash: Bytes! # bytes32
+  recipient: Bytes! # address
+  fillReceiptMakerToken: Bytes! # address
+  fillReceiptTakerToken: Bytes! # address
+  fillReceiptMakerTokenFilledAmount: BigInt! # uint256
+  fillReceiptTakerTokenFilledAmount: BigInt! # uint256
+  fillReceiptRemainingAmount: BigInt! # uint256
+  fillReceiptMakerTokenFee: BigInt! # uint256
+  fillReceiptTakerTokenFee: BigInt! # uint256
+  blockNumber: BigInt!
+  blockTimestamp: BigInt!
+  transactionHash: Bytes!
+}
+
+type OrderCancelled @entity(immutable: true) {
+  id: ID!
+  orderHash: Bytes! # bytes32
+  maker: Bytes! # address
+  blockNumber: BigInt!
+  blockTimestamp: BigInt!
+  transactionHash: Bytes!
+}
+
+enum OrderStatus {
+  Normal
+  Cancelled
+  FullyFilled
+}
+
+enum LimitOrderTypes {
+  ByProtocol
+  ByTrader
+}
+
+type Order @entity(immutable: false) {
+  id: ID! # orderHash
+  limitOrderFilledId: [String!]! # One-to-many relationship
+  orderStatus: OrderStatus!
+  maker: Bytes! # address
+  makerToken: Bytes! # address
+  takerToken: Bytes! # address
+  firstFilledTime: BigInt!
+  lastFilledTime: BigInt!
+  cancelledTime: BigInt!
+}
+
+type LimitOrder @entity(immutable: true) {
```

Comment:
Seems like this `LimitOrder` entity is the intersection of `LimitOrderFilledByProtocol` and `LimitOrderFilledByTrader`. Wondering what's the purpose to have another similar entity here?

Reply:
### LimitOrder Entity
Yes! the LimitOrder Entity is the union of LimitOrderFilledByTrader and LimitOrderFilledByProtocol.
It was designed this way to allow developers to directly query the LimitOrder Entity to obtain all Filled data.

Additionally, there are the OrderCancelled Entity and Order Entity.
### OrderCancelled Entity
The OrderCancelled Entity collects event parameters sent by the cancelLimitOrder function.

### Order Entity
- The Order Entity and the LimitOrder Entity have a one-to-many relationship with each other, and they are linked by orderID and eventID (similar to a one-to-many record in a relational database).
- The Order Entity stores the Order information created by the maker, including the current status of the Order (Normal, Cancelled, FullyFilled) and the IDs of any Filled (ByTrader, ByProtocol) that interacted with the Order.
- It also saves the timestamp of the first and last appearance of the Order. This allows developers to query the status of their interested Orders.